CMake/CurlTests.c

```diff
@@ -318,6 +318,29 @@ int main(void)
 }
 #endif
 
+#ifdef HAVE_SETSOCKOPT_SOL_IP
```

Comment:
It'd be useful to pre-cache this for Windows, which always seems to fail this test:
```diff
--- a/CMake/Platforms/WindowsCache.cmake
+++ b/CMake/Platforms/WindowsCache.cmake
@@ -170,6 +170,7 @@ set(HAVE_IOCTLSOCKET_CAMEL_FIONBIO 0)
 set(HAVE_IOCTLSOCKET_FIONBIO 1)
 set(HAVE_IOCTL_FIONBIO 0)
 set(HAVE_SETSOCKOPT_SO_NONBLOCK 0)
+set(HAVE_SETSOCKOPT_SOL_IP 0)
 set(HAVE_POSIX_STRERROR_R 0)
 set(HAVE_BUILTIN_AVAILABLE 0)
 set(HAVE_MSG_NOSIGNAL 0)
```

Reply:
Nice, thanks!